platform/mds/mds-web/src/main/java/org/motechproject/mds/web/service/impl/InstanceServiceImpl.java

```diff
@@ -536,23 +539,33 @@ private void updateFields(Object instance, List<FieldRecord> fieldRecords, Motec
         }
     }
 
-    private List<EntityRecord> instancesToRecords(Collection instances, EntityDto entity, List<FieldDto> fields,
+    private List<BasicEntityRecord> instancesToBasicRecords(Collection instances, EntityDto entity, List<FieldDto> fields,
                                                   MotechDataService service) {
-        List<EntityRecord> records = new ArrayList<>();
+        List<BasicEntityRecord> records = new ArrayList<>();
         for (Object instance : instances) {
-            EntityRecord record = instanceToRecord(instance, entity, fields, service);
+            BasicEntityRecord record = instanceToBasicRecord(instance, entity, fields, service);
             records.add(record);
         }
         return records;
     }
 
+    private BasicEntityRecord instanceToBasicRecord(Object instance, EntityDto entityDto, List<FieldDto> fields,
+                                                    MotechDataService service) {
+        return instanceToRecord(instance, entityDto, fields, service, true);
+    }
+
     private EntityRecord instanceToRecord(Object instance, EntityDto entityDto, List<FieldDto> fields,
                                           MotechDataService service) {
+        return (EntityRecord) instanceToRecord(instance, entityDto, fields, service, false);
+    }
+
+    private BasicEntityRecord instanceToRecord(Object instance, EntityDto entityDto, List<FieldDto> fields,
```

Comment:
make this a generic <T> method that returns T and takes a Class<T extends BasicRecord> parameter, that's what I meant. That way it will know what T is from the class parameter. Many methods in libraries do that